common-api/src/main/java/org/jboss/galleon/util/IoUtils.java

```diff
@@ -152,10 +152,26 @@ public static void copy(Path source, Path target) throws IOException {
     }
 
     public static void copy(Path source, Path target, boolean skipExistingFiles) throws IOException {
-        if(Files.isDirectory(source)) {
-            Files.createDirectories(target);
+        if (Files.isDirectory(source)) {
+            try {
+                Files.createDirectories(target);
+            } catch (FileAlreadyExistsException e) {
+                // FileAlreadyExistsException happen if the target path already exists and it's not directly (symbolic link or regular file)
+                // this exception can be ignored if it's symbolic link and if real path of the symbolic link is directory.
+                if (!Files.isDirectory(target)) {
+                    throw new IOException("Unable to copy file(s) from source path [" + source + "] to target path [" + target + "] because the target path is not directory (regular file or symbolic link where real path is not directory).", e);
```

Comment:
I don't fully follow the content of the Exception. Should it be: " because the target path is not a directory (it is a regular file or a symbolic link where the target path is not a directory)"?